test/Test/Mockery/LoggingSpec.hs

```diff
@@ -0,0 +1,32 @@
+module Test.Mockery.LoggingSpec (main, spec) where
+
+import Test.Hspec
+import Data.IORef
+import System.Logging.Facade.Types
+import System.Logging.Facade.Sink
+import System.Logging.Facade as Log
+
+import Test.Mockery.Logging
+
+main :: IO ()
+main = hspec spec
+
+spec :: Spec
+spec = describe "Test.Mockery.Logging" $ do
+  describe "captureLogs" $ do
+    let logToIORef :: IORef [LogRecord] -> LogSink
+        logToIORef ref record = modifyIORef ref (record :)
+
+    it "returns all logs of an action" $ do
+      (logs, ()) <- captureLogs $ do
+        Log.trace "this should be captured"
+        Log.trace "this should be captured next"
+      logs `shouldBe` [ (TRACE, "this should be captured")
+                      , (TRACE, "this should be captured next")
+                      ]
+
+    it "prevents logs from going to the log sink" $ do
+      ref <- newIORef []
+      setLogSink $ logToIORef ref
+      _ <- captureLogs $ Log.trace "this should be captured"
+      readIORef ref `shouldReturn` []
```

Comment:
I would probably not test this, for the reason that I think it would take extra effort to implement it differently.

What I find more interesting is that

    it "restores the original log sink"